### ISSUE TITLE
Running lrp4rag on other models

### ISSUE THREAD

**Issue by Vanessa-Taing** [Tomsawyerhu/LRP4RAG#2]
Hi,

I am trying to evaluate other models with lrp4rag. I do have a few questions regarding the operation:

1. Which model is used for the lrp evaluation, as in line 9 of llama_lrp.py: `model_id = '/Users/tom/Downloads/mini-llama'` . I am unable to find the specific model on HuggingFace, am I right to assume that all llama supported models can work in that case? What is the impact of choosing between a 2B/7B model for this evaluation?

2. I tried running the code on the llama 7B dataset with modifications:
```
model_id = 'akjindal53244/Llama-3.1-Storm-8B-FP8-Dynamic'
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
max_length = 100000
prompt = "",
model = AutoModelForCausalLM.from_pretrained(model_id, local_files_only=True)
tokenizer = AutoTokenizer.from_pretrained(model_id)
attnlrp.register(model)
```
The output was:
```
1th token, token=<|begin_of_text|>, rank=105542
Killed
```
And I noticed the loading of shards was taking a few minutes, which was quite long as compared to my other attempts in loading a model.
May I know what is inhibiting the process from completing?

3. In my case where I want to test a model available on HuggingFace: [akjindal53244/Llama-3.1-Storm-8B](akjindal53244/Llama-3.1-Storm-8B), from my understanding I will have to test the model on the RAGTruth (QA part) to obtain a dataset of hallucination/nonhallucination in jsonl format, then upload to the data directory to be evaluated. Is this the correct way of doing so? If yes, what is an effective way of preparing the dataset? (I have no prior experience in this kind of task)

Thank you for the impressive work, really appreciate if you could help.

**Comment by Tomsawyerhu** [comment]
mini-llama is just a small model which is used for test. Actually, when doing lrp back propogation, we use llama-2-7b-chat and llama-2-13b-chat consistent  with RAGTruth setting. You may also use llama3 to simulate the distribution which we have not tried yet, because RAGTruth do not have samples for llama-3. Since we cannot get RAGTruth experiment settings, so the first token's logits may seem wired, you can take it as a different sampling process which does no harm to result.

**Comment by Tomsawyerhu** [comment]
To save your time, you can follow readme to download lrp result.

**Comment by Vanessa-Taing** [comment]
Hi, was able to run the classifier code using the downloaded lrp result for llama2 13b:

```
python classifier.py --datasets ../lrp_result_llama_13b --classifier SVM
----------------------------------------------------------------------------------------------------------
Training Accuracy: 82.30%
Training Precision: 89.30%
Training Recall: 62.14%
Validation Accuracy: 58.59%
Precision: 57.69%
Recall: 33.33%
F1: 42.25%
Training Accuracy: 82.55%
Training Precision: 83.46%
Training Recall: 70.25%
Validation Accuracy: 63.13%
Precision: 56.41%
Recall: 53.01%
F1: 54.66%
Training Accuracy: 81.92%
Training Precision: 84.41%
Training Recall: 68.52%
Validation Accuracy: 66.16%
Precision: 57.14%
Recall: 42.67%
F1: 48.85%
Training Accuracy: 81.42%
Training Precision: 85.16%
Training Recall: 66.67%
Validation Accuracy: 66.16%
Precision: 54.24%
Recall: 44.44%
F1: 48.85%
Training Accuracy: 82.83%
Training Precision: 86.51%
Training Recall: 68.12%
Validation Accuracy: 61.93%
Precision: 52.86%
Recall: 46.84%
F1: 49.66%
avg accuracy=0.6319386761011127,avg precision=0.5566797044763147,avg recall=0.44058387135038046, avg f1=0.48857251882850344
```

I am assuming these are the results for llama2 13b. In that case, I will need to get llama3 to perform RAGTruth simulation before using this part of the code, is that a correct way of thinking?

Thank you.

**Comment by Tomsawyerhu** [comment]
However, currently there no hallucinated samples for llama3 in RAGTruth dataset, you may annotate samples yourself or look for other datasets. Btw,  SVM is not very efficient for llama-2-13b, you may try lstm or enlarge C and gamma for SVM to suit a complicated classification.
